test/example/generated/ExampleDialect.cpp.inc

```diff
@@ -640,6 +640,59 @@ source,
           errs << "    actual:    " << printable(::llvm::cast<::llvm::FixedVectorType>(sourceType)->getNumElements()) << '\n';
           return false;
         }
+      if (![&]() {
+
+      {
+        std::array<std::string, 2> orErrors;
+        if (true
+    
+      && !([&]() {
+        ::llvm::raw_string_ostream errs(orErrors[0]);
+    
+        if (xd::VectorKind::LittleEndian != ::llvm::cast<XdVectorType>(resultType)->getKind()) {
+          errs << "  inconsistent value of eq:$rhs found\n";
+          errs << "  while checking VectorKindLittleEndian:\n";
+          errs << "    here:       " << printable(xd::VectorKind::LittleEndian) << '\n';
+          errs << "    previously: " << printable(::llvm::cast<XdVectorType>(resultType)->getKind()) << '\n';
+          return false;
+        }
+      return true;
+}())
+
+      && !([&]() {
+        ::llvm::raw_string_ostream errs(orErrors[1]);
+    
+        if (xd::VectorKind::BigEndian != ::llvm::cast<XdVectorType>(resultType)->getKind()) {
+          errs << "  inconsistent value of eq:$rhs found\n";
+          errs << "  while checking VectorKindBigEndian:\n";
+          errs << "    here:       " << printable(xd::VectorKind::BigEndian) << '\n';
+          errs << "    previously: " << printable(::llvm::cast<XdVectorType>(resultType)->getKind()) << '\n';
+          return false;
+        }
+      return true;
+}())
+) {
```

Comment:
Oh, you mean because usually, only one of the options applies, and so even if the verification is successful it's possible that we create the strings for some of the options, depending on how the short-circuiting works out?

That's a fair point. I suppose we could create the verifier code for the options twice, once without output for the check and then once with output for the error printing.